### PR TITLE
refactor: transform to native classes via codemod

### PR DESCRIPTION
**Pull request by derrabauke**
## Contents:
* 388ee9d refactor(ember-tracked-properties-codemod): transform all classes
* 10fcd48 refactor(ember-native-class-codemod): components
* 39218cb refactor(ember-native-class-codemod): services
* bfa5eaa refactor(no-get-with-defaults-codemod): replace getWithDefaults

The contents of this PR start at bfa5eaa
Commits before belong to #715

---

---

### CODE REVIEW

File: app/components/in-viewport/component.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
     const observer = new IntersectionObserver(
       ([{ isIntersecting }]) => {
         if (isIntersecting) {
-          return this.getWithDefault("on-enter-viewport", () => {})();
+          return (get(this, "on-enter-viewport") ?? (() => {}))();
         }
 
-        return this.getWithDefault("on-exit-viewport", () => {})();
+        return (get(this, "on-exit-viewport") ?? (() => {}))();
```
Comment (Yelinz) [inline app/components/in-viewport/component.js:20]: Also very weird

File: app/components/sort-header/component.js [hunk 2]
```diff
@@ -8,19 +10,21 @@ const getDirection = state => {
...
+export default class SortHeader extends Component {
+  @(computed("current").readOnly())
+  get direction() {
     return getDirection(this.current);
-  }).readOnly(),
+  }
 
-  active: computed("current", function() {
+  @(computed("current").readOnly())
+  get active() {
```
Comment (Yelinz) [inline app/components/sort-header/component.js:22]: No need for the computed with readonly?

File: app/components/sy-durationpicker/component.js [hunk 3]
```diff
@@ -20,18 +17,16 @@ const { abs } = Math;
...
-
-  max: MAX_SAFE_INTEGER,
-
-  maxlength: null,
+@classic
+export default class SyDurationpicker extends SyTimepickerComponent {
+  name = "duration";
+  @tracked min = MIN_SAFE_INTEGER;
+  max = MAX_SAFE_INTEGER;
+  maxlength = null;
```
Comment (Yelinz) [inline app/components/sy-durationpicker/component.js:25]: Only 1 tracked?

File: app/components/sy-timepicker/component.js [hunk 4]
```diff
@@ -87,38 +74,38 @@ export default Component.extend({
    * @property {Number} precision
    * @public
    */
-  precision: 15,
+  @tracked precision = 15;
```
Comment (Yelinz) [inline app/components/sy-timepicker/component.js:77]: Any reason why only this property is `tracked`?

File: app/components/welcome-modal/component.js [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
-export default Component.extend({
-  tagName: "",
-  "on-never"() {},
-  "on-later"() {},
+@classic
+@tagName("")
+export default class WelcomeModal extends Component {
+  "on-never"() {}
+  "on-later"() {}
   "on-start"() {}
```
Comment (Yelinz) [inline app/components/welcome-modal/component.js:10]: These are some wild function names...

File: app/services/metadata-fetcher.js [hunk 6]
```diff
@@ -41,7 +44,7 @@ export default Service.extend({
    * @return {Object} An object with the parsed metadata
    * @public
    */
-  fetchSingleRecordMetadata: task(function*(type, id) {
+  @(task(function* (type, id) {
```
Comment (Yelinz) [inline app/services/metadata-fetcher.js:47]: Quick refactor to restartableTask

File: app/services/tracking.js [hunk 7]
```diff
@@ -119,18 +127,16 @@ export default Service.extend({
    * @method _computeTitle
    * @private
    */
-  _computeTitle: task(function*() {
-    while (this.get("activity.active")) {
-      const duration = moment.duration(
-        moment().diff(this.get("activity.from"))
-      );
+  @task(function* () {
```
Comment (Yelinz) [inline app/services/tracking.js:130]: Move function out of decorator

**Review by Yelinz — APPROVED** [review]
Just some stylistic things, no need to change as long as it works

**Comment by derrabauke** [comment]
> Just some stylistic things, no need to change as long as it works

Thanks for the feedback! Will incorporate this after the automated codemods are done!

**Pull request merged**